syntax = "proto3";

package grpcSkill;

service grpcSkill {
    rpc RegisterCurrentInstance(RegisterRequest) returns(RegisterResponse) {}
    rpc DetermineSkillFromMessage(SkillDeterminationRequest) returns(SkillDeterminationResponse) {}
}

message SkillDeterminationResponse {
    float Accuracy = 1;
    float Duration = 2;
    string FowardAddress = 3;
}
 
message SkillDeterminationRequest {
    string Message = 1;
}

message SkillInfo {
    string ExampleText = 1;
    string POSMapping = 2;
}

message RegisterRequest {
    repeated SkillInfo ExampleInfo = 1; 
    string SkillName = 2;
}

message RegisterResponse {
    bool IsRegistered = 1;
}